internal/core/queries/decode_vin.go

```diff
@@ -106,8 +106,10 @@ func (dc DecodeVINQueryHandler) Handle(ctx context.Context, query mediator.Messa
 		return nil, errors.Wrap(err, "error when beginning transaction")
 	}
 	defer txVinNumbers.Rollback() //nolint
-
-	vinDecodeNumber, err := models.FindVinNumber(ctx, txVinNumbers, vin.String())
+	vinDecodeNumber, err := models.VinNumbers(
+		models.VinNumberWhere.Vin.EQ(vin.String()),
+		qm.Load(models.VinNumberRels.DeviceDefinition)).
+		One(ctx, dc.dbs().Reader)
```

Comment:
Should we keep using `txVinNumbers` here?

Reply:
I was wondering if we need a transaction at all here since we are checking for existence and creating if it doesn't exist. Since there is nothing to rollback if a create fails.

Reply:
It doesn't hurt so I will just use it, but I am curious if you can think of an edge case I am missing.